app/src/main/java/com/isc/hermes/model/Utils/IncidentsUtils.java

```diff
@@ -0,0 +1,59 @@
+package com.isc.hermes.model.Utils;
+
+import org.bson.types.ObjectId;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+/**
+ * Utility class for handling incidents-related operations.
+ */
+public class IncidentsUtils {
+    /**
+     * Generates a new ObjectId as a hexadecimal string.
+     *
+     * @return The generated ObjectId as a hexadecimal string.
+     */
+    public static String generateObjectId() {
+        ObjectId objectId = new ObjectId();
+        return objectId.toHexString();
+    }
+    /**
+     * Adds a specified time duration to the current date and returns it as a formatted string.
+     *
+     * @param timeDuration The time duration to add in the format "X unit" (e.g., "2 hours", "1 day").
+     * @return The modified date as a formatted string.
+     */
+    public static String addTimeToCurrentDate(String timeDuration) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+
+        String[] parts = timeDuration.split(" ");
+        int amount = Integer.parseInt(parts[0]);
+        String unit = parts[1].toLowerCase();
+
+        if (unit.equals("minute") || unit.equals("minutes")) {
+            calendar.add(Calendar.MINUTE, amount);
+        } else if (unit.equals("hour") || unit.equals("hours")) {
+            calendar.add(Calendar.HOUR, amount);
+        } else if (unit.equals("day") || unit.equals("days")) {
+            calendar.add(Calendar.DAY_OF_YEAR, amount);
+        } else if (unit.equals("month") || unit.equals("months")) {
+            calendar.add(Calendar.MONTH, amount);
+        }
```

Comment:
Nested if is not a good practice. You could use a map to store the values and access to them. You can review the 
following suggestion. Of course the map could be a class attribute to not create it every time the method is being called.

```suggestion
        Map<String, Integer> unitMap = new HashMap<>();
        unitMap.put("minute", Calendar.MINUTE);
        unitMap.put("minutes", Calendar.MINUTE);
        unitMap.put("hour", Calendar.HOUR);
        unitMap.put("hours", Calendar.HOUR);
        unitMap.put("day", Calendar.DAY_OF_YEAR);
        unitMap.put("days", Calendar.DAY_OF_YEAR);
        unitMap.put("month", Calendar.MONTH);
        unitMap.put("months", Calendar.MONTH);
  
        Integer unitValue = unitMap.get(unit);
        if (unitValue != null) {
            calendar.add(unitValue, amount);
        }
```